### DISCUSSION TITLE
v2.7.2

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#4233] [分享 / Share]
> ⚠ 对浏览器的要求中, Safari 提升至 15.4

✨新增

<details>
<summary>获得 v2.6.1-preview ~ v2.7.1-preview 的所有新功能, 点击展开查看</summary>

- `自定义顶栏` 搜索优化:
  - 提高了搜索 av 号 / BV 号结果的优先级, 输入编号后直接回车就可以前往视频.
  - 支持搜索多种 ID, 可以通过安装对应的插件来使用: (#677)
    - 搜索 av / BV 号时, 支持显示视频标题
    - 支持搜索专栏 cv 号 / 文集 rl 号
    - 支持搜索音频 au 号 / 播放列表 am 号
    - 支持搜索番剧 ss / ep 号, 或番剧详情 md 号
    - 支持搜索用户 uid 号
    - 纯数字输入时, 支持同时搜索视频和专栏
- `自定义顶栏` 的收藏和稍后再看支持显示数量. (#4069)
- `自定义顶栏` 的订阅支持过滤 在看 / 看过 / 想看. (#3217)
- 添加跨域存储功能 (`crossOriginLocalStorage`), 支持搜索历史跨域同步. (#3893)
- 新增功能 `弹幕转义`. (#3470)
> 将弹幕中的 `\n` 替换为真实的换行, 注意这可能导致原先不重叠的弹幕发生重叠.
- 新增功能 `组件二等分`, 在出现问题时可帮助定位是哪个组件导致的问题. (#3829, PR #3965 by [JLoeve](https://github.com/LonelySteve))
- 优化了 `网址参数清理` 的逻辑, 通过 Hook History API 来减少出现重复的历史记录. (#4039)
- 新增功能 `隐藏热搜`. (#3744)
> 隐藏搜索栏和搜索页面中的 `bilibili 热搜`.
- 新增功能 `隐藏视频分享`. (#3663)
> 隐藏视频和番剧播放器下方的分享按钮.
- 新增插件 `快捷键扩展 - 开关灯`. (#3587)
> 在快捷键的动作列表里添加一个 "开关灯".
- `显示视频投稿时间` 允许自定义替换文本格式. (PR #4136 by [呼乎户](https://github.com/wisokey))
- `清爽首页` 排行榜支持切换为列表模式, 能够查看 10 项左右的排行. (#3611, #3175)
- 新增功能 `动态看图自动回顶`. (#4029)
> 在动态里查看图片详情时 (非全屏), 切换图片自动回到图片顶部; 退出查看图片模式时, 自动将动态移入视图内.
- 重构了 `直播信息扩充` 面板, 改为完全自主实现, 支持显示超过 10 条的直播间, 并可以进行搜索和刷新操作. (#3774, #4035, #4034)

<img src="https://user-images.githubusercontent.com/26504152/234049176-7ae3e3cf-6956-4efe-aec1-66218d940c93.png" width="300">

- 新增功能 `反转滚轮调音量`. (#4053)
> 反转在网页全屏 / 全屏模式下使用滚轮调节音量的方向, 使其更符合使用触控板时的操作方向. 请注意不能和 `禁止滚轮调音量` 一同使用.
> - 手指向上推时, 增加音量
> - 手指向下推时, 减少音量
> - 可以自定义 `灵敏度`, 同样的滚动幅度下, 灵敏度越高变化的音量越多

- 新增功能 `自动点赞`. (#2354)
> 进入视频 / 查看动态时, 自动点赞.

- 新增功能 `隐藏用户信息卡片`. (PR #4093 by [WakelessSloth56](https://github.com/WakelessSloth56))
> 隐藏鼠标指向用户名或用户头像时弹出的浮动用户信息卡片.

- 新增功能 `播放器置顶（新）`. (#3030, #PR #4152 by [RieN 7z](https://github.com/rien7))
> 原来的播放器置顶组件，现在已经不可用了，这是一个新的版本，可以在视频页面中将播放器放在页面最上方.

- 启用新的代码加载方式. (PR #3938 by [timongh](https://github.com/timongh))
- Comments API 支持读取评论图片. (PR #4130 by [manip_ego](https://github.com/manipEgo))
```diff
export interface CommentItem extends CommentReplyItem {
+  /** 评论图片 */
+  pictures?: string[]
  /** 回复 */
  replies: CommentReplyItem[]
  // ...
}
```

</details>

- `网址参数清理` 增加清理会员购和直播间的一些参数. (PR #4189, PR #4214 by [星海](https://github.com/lovelyCARDINAL))
- `删除广告` 增加选项 `保留小喇叭通知`. (PR #4199 by [imshixin](https://github.com/imshixin))
> 指评论区上方的小喇叭通知, 待 `简化评论区` 完成功能拆分后会移动到那边去.

- `与 "解除 B 站区域限制" 互斥` 现在不再在番剧详情页面生效. (#4157)

🐛修复
- 修复播放器模式切换时 `body` 类名响应问题. (PR #4188 by [Ziu](https://github.com/ZiuChen))
- 修复 `播放器控制栏背景色` 位置不正确. (#3713)
- 修复勋章过多时, `直播勋章快速更换` 的弹窗溢出视图. (#4181)
- 修复 `清爽首页` 在非内测首页上无法隐藏原版首页. (#3981)
- 修复 `自定义顶栏` 导致分区页面内容错位. (#4183)
- 修复 `删除广告` 未能删除活动横幅. (#4177)
- 修复 `外置稍后再看` 没有隐藏菜单中的相同选项. (#4175)
- 修复 `直播全屏包裹` 的一部分样式问题. (#4166, #4210)
- 修复检查更新的查看详情弹窗内容溢出. (#4164)
- 修复 `播放前显示封面` 在预览关联视频时封面消失. (#4197)
- 修复 `禁止跳转动态详情` 对新版专栏卡片无效.
- 修复 `动态过滤器` 在切换了 UP 主过滤后失效. (#4149)
- 修复 `简化直播间` 的 `标题栏活动` 内容出现溢出. (#4215)
- 新版动态页面适配:
  - `快速收起评论区`
  - `addMenuItem` API (涉及 `图片批量导出` 和 `复制动态链接`)
  - `强制固定动态侧栏`: 由于 class 名称和旧版一样, 目前只能先把宽度也锁死在旧版的宽度.
  - `夜间模式`
  - `feedsCardsManager` API

☕开发者相关
- GitHub 相关配置更换为 JSON 格式, 并由脚本自动生成 YAML.
- 增加了讨论区 `功能建议` 的模板.

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.7.2'>v2.7.2</a>.</em>

**Comment by the1812** [comment]
(讨论区那个模板好像没生效, 不知道这文件名到底咋匹配的...

**Comment by pinaZC** [comment]
https://github.com/the1812/Bilibili-Evolved/issues/4215#issue-1722143807
这个的第二个问题还没适配啊，弹窗还是显示不全，无法关闭；能不能重新开启下这个问题，要么适配好，要么直接删掉这个助力按钮，也算简化了

**Reply by pinaZC** [reply]
建议还是留着这个按钮，做好适配，毕竟是b站新的统计主播人气的方式，还算是观众直接支持主播的一个方式
